### ISSUE TITLE
Corrupted zip: Missing bytes

### ISSUE THREAD

**Issue by shyamcharit03** [guigrpa/docx-templates#116]
I'm trying out the module and executing a basic scenario. Please see code snippet below. I get the following error. I'm executing this on Node version 10.18.0 and on Ubuntu 18.04.

**Error:**

Error: Corrupted zip: missing 528 bytes.
    at ZipEntries.readEndOfCentral (/home/shyam/source/src/slyncer/node_modules/docx-templates/node_modules/jszip/lib/zipEntries.js:244:19)
    at ZipEntries.load (/home/shyam/source/src/slyncer/node_modules/docx-templates/node_modules/jszip/lib/zipEntries.js:256:14)

**Code snippet**

const template = await readHTMLFile('/../../data/transactionTemplates/DATED.docx');

  const buffer = await docxTemplate.createReport({
    template,
    data: {
      date: '01/01/2020',
    },
  });

  res.setHeader(
    'Content-Type',
    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
  );
  res.setHeader('Content-Disposition', 'attachment; filename=test.docx');
  res.setHeader('Content-Length', buffer.length);
  res.send(buffer);

**Comment by jjhbw** [comment]
Can you give me a template file to reproduce this with? The error seems to indicate a corrupt template.

Also, i dont know what is happing in your readHTMLFile function. If this func alters the docx file in some way after reading then jsZip likely wont be able to use it. Note that the input template needs to be an unmodified Buffer or Buffer-like representation of the docx file.

**Comment by shyamcharit03** [comment]
Hi,

Thank you for pointing me in the right direction. The readHTMLFile function was some generic function in the codebase that hadn't been touched for a longtime and I did not bother to look at it. The issue was resolved after I removed utf-8 encoding from the options. Thanks a lot for your help.

let readHTMLFile = async function(filePath) {
  try {
    return fs.readFileSync(__dirname + filePath, {
      encoding: 'utf-8',
    });
  } catch (err) {
    logger.error(`Error: readHTMLFile(), Message: ${err.message}, Stack: ${err.stack}`);
    throw err;
  }
};
